lib/active_model/serializer/collection_serializer.rb

```diff
@@ -46,7 +46,10 @@ def json_key
         # 3. get from collection name, if a named collection
         key ||= object.respond_to?(:name) ? object.name && object.name.underscore : nil
         # 4. key may be nil for empty collection and no serializer option
-        key && key.pluralize
+        key &&= key.pluralize
+        # 5. fail if the key cannot be determined
+        key || fail(ArgumentError, 'Cannot infer root key from collection type. Please
```

Comment:
I would have thought this required `or`.  This is fine

Reply:
FYI the only difference between `||` and `or` is operator precedence: `a = false || true` is equivalent to `(a = false) || true`.